### DISCUSSION TITLE
Speech Background Noise Suppression with Deep Learning (Project 193)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#24] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #193](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Speech%20Background%20Noise%20Suppression%20with%20Deep%20Learning)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by pyshx** [comment]
Hello Universe!!

**Reply by robertogl** [reply]
Hello to you, Piyush! Great to know you will be working on this project. Do not hesitate to write any comments or questions related to the project.

**Reply by jibrahim80** [reply]
Great to see you are working on this project, Piyush! Please do share questions/comments here!

**Comment by akshat47-glitch** [comment]
Hi, i am Akshat. Excited to be working with y'all!

**Reply by akshat47-glitch** [reply]
hey, where's the mathworks repository mentioned in the mail

**Reply by jibrahim80** [reply]
Hi Akshat,

Which repository are you referring to?

**Reply by robertogl** [reply]
Welcome Akshat!

The repository mentioned in the email is this one, with the projects. You do not really need to download the whole repository, but to have the project description available even when you are offline you may download just the readme file, but that is up to you.  Sorry for any confusion.

**Comment by lucy0828** [comment]
Hi! I'm Ahhyun Yuh and currently on my masters at Kyungpook National University Real-Time System Lab. I research on Sound Event Detection and I would like to participate in your project.

**Comment by BanmaS** [comment]
Hello, we are undergraduates from Shanghai Jiaotong University. We are honored to be able to participate in your project as a project of our course "Intelligent System Design and Practice"!
Regarding your project, our idea is to do basic suggested steps based on existing algorithms, then do comparative experiments in project variations, and finally consider advanced project work. We want to know whether such a route meets the requirements of your project.

**Reply by jibrahim80** [reply]
Hi BanmaS! Great to see you are interested in this project. 
Yes, I think your suggested approach is a good and valid  way to work on this project.

**Reply by BanmaS** [reply]
Thank you very much! And we still have a question about the fourth item in advanced projects, "Consider personalized noise suppression, where your network is trained to improve the speech quality of a particular speaker. In this scenario, you have access to a few minutes of speech from this speaker, from which you can extract features in the training stage. "We don’t understand this specific requirement very well. Does it refer to a person in a noisy scene where many people speak, to improve the clarity of his speech? And intensity, while suppressing the voice of other people talking?

**Reply by jibrahim80** [reply]
Hi BanmaS, what is meant by "personalized noise suppression" is noise suppression that only works well for a specific person (and is intended to work well for any random person). If you are training a network to denoise the speech of one person in particular, that network can potentially outperform a more general network that has to do a good job for any person. So the scenario is the same as other cases, i.e. just one person speaking with noise, and the goal is to denoise the speech, but hopefully specializing the network on one speaker will give superior results for that one speaker. The drawback is that you will need access to training data from that particular speaker, but that is a reasonable assumption for many realistic use cases.

**Comment by ashu1069** [comment]
Hey, I'm Ashutosh from IIT Roorkee. Glad to be a part of this project.
I need to ask if the usage of the dataset is restricted to the one given in the project description or can we borrow that from other sources too such as https://datashare.ed.ac.uk/handle/10283/1942 ?

**Reply by jibrahim80** [reply]
Hi Ashutosh,

Nice to see you're working on this project! I think it is fine to use extra data from other sources to train your model.

**Reply by ashu1069** [reply]
Thank you for the clarification!

**Comment by lohithareddy2511** [comment]
Hey! I hope this message finds you well. I am currently working on Project 193: Speech Background Noise Suppression with Deep Learning, and I had a couple of queries regarding the dataset and implementation process.

I have followed the provided link to the Microsoft DNS Challenge repository to download the dataset. However, the unpacked dataset appears to be extremely large in size, and I am facing challenges in terms of download time, storage, and handling. Could you please provide guidance on:

1. Whether there is a smaller or recommended subset of the dataset for initial training and experimentation purposes?
2. Any specific instructions or scripts for efficiently downloading, organizing, and preprocessing the dataset for use with MATLAB?
3. A suggested workflow or step-by-step process to successfully implement and test the deep learning-based noise suppression model using the Audio Toolbox and Deep Learning Toolbox?

Thank you for your support, and I look forward to your response.

Best regards,
Lohitha Reddy Karamala

**Comment by Jvc-largo** [comment]
Hello, I am a student at UBC and I am interested in working and collaborating on the "Speech Background Noise Suppression with Deep Learning" project!
